compiler/rustc_passes/src/check_attr.rs

```diff
@@ -619,7 +619,30 @@ impl CheckAttrVisitor<'_> {
             }
             // we check the validity of params elsewhere
             Target::Param => return false,
-            _ => None,
+            Target::Expression | Target::Statement | Target::Arm => Some(target.name()),
+            Target::ExternCrate
+            | Target::Use
+            | Target::Static
+            | Target::Const
+            | Target::Fn
+            | Target::Closure
```

Comment:
A closure has no name, this needs to go in the arm you added.

Reply:
Resolved by 2be0094